### PR TITLE
Rewrote the getType method to handle all ploidy levels

### PR DESCRIPTION
**Pull request by NeillGibson**
Hi, 

I rewrote the getType method to handle all ploidy levels. 

This is inspired by getType / determineType method in HTS-JDK genotype:
https://github.com/samtools/htsjdk/blob/bd92747fd3672635de96473fea6d4b38e8635c8e/src/java/htsjdk/variant/variantcontext/Genotype.java#L197

I am not sure if the code runs trough the tests, I have some difficulty setting up all code dependencies locally in my IDE (IntelliJ) to run the tests. If the code fails tests please see this as the pseudo code of the method. 

One open issue for this code is still which genotypeType to return for genotypes that consist of multiple OtherAlt alleles.  https://github.com/bigdatagenomics/adam/issues/897

Currently the method in this pull request returns NoCall for a genotype consisting of just OtherAlt alleles.  This is also what the current method does.

### CODE REVIEW

**Comment by heuermh** [comment]
Jenkins, test this please.

**Comment by NeillGibson** [comment]
Jenkins, test this please.

**Comment by heuermh** [comment]
Jenkins, retest this please.

File: adam-core/src/main/scala/org/bdgenomics/adam/rich/RichGenotype.scala [hunk 1]
```diff
@@ -26,16 +26,56 @@ object RichGenotype {
...
+        // If genotype.getAlleles returns a multiple OtherAlt, the genotype is
+        // A) The OtherAlt alleles are the same OtherAlt alleles: Homozygous Alternative (HOM_ALT)
+        // B) The OtherAlt allales are different OtherAlt alleles: Heterozygous
+        // For now return NO_CALL as the genotypes, as was done in the previous getType function
+        // See also issue https://github.com/bigdatagenomics/adam/issues/897
+        case List(GenotypeAllele.OtherAlt) => GenotypeType.NO_CALL
+
+        // only the four above alleles are possible
+        // https://github.com/bigdatagenomics/bdg-formats/blob/master/src/main/resources/avro/bdg.avdl#L464
+        case _                             => throw new Exception("This should be impossible. Found single distinct allele other than the four possible alleles: Ref, Alt, NoCall and OtherAlt  ")
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rich/RichGenotype.scala]: use [IllegalStateException](http://docs.oracle.com/javase/7/docs/api/java/lang/IllegalStateException.html) instead?

Then the message could just be ```"Found single distinct allele other than the four possible alleles: Ref, Alt, NoCall and OtherAlt"```

Comment (NeillGibson) [inline adam-core/src/main/scala/org/bdgenomics/adam/rich/RichGenotype.scala]: Yes thank you for the advice. I'll update this exception in a new commit accordingly.

**Comment by NeillGibson** [comment]
I've squashed the commits in to a single commit.

**Comment by fnothaft** [comment]
Thanks @NeillGibson! Committed as 30f937fc177ff3511129bcb3eaec4c987b851f4a.

**Pull request closed without merging**